lib/ramble/ramble/application.py

```diff
@@ -829,7 +829,19 @@ def build_modifier_instances(self):
             self._modifier_instances.append(mod_inst)
 
             # Add this modifiers required variables for validation
-            self.keywords.update_keys(mod_inst.required_vars)
+            required_vars = mod_inst.required_vars
```

Comment:
I think I'd prefer if we push the logic to extract required variables into the modifier rather than having the application class pull them out directly.

Reply:
Good point, updated.